cypress/integration/products_page.spec.js

```diff
@@ -1,12 +1,102 @@
 describe('My First Test', () => {
 
-  // 1. Visit a web page.
-  // 2. Query for an element.
-  // 3. Interact with that element.
-  // 4. Assert about the content on the page.
- 
-  it('Visits the products page', () => {
-    cy.visit('http://localhost:4200/');
-  });
+  describe('basket', () => {
+
+    it('should visit the product page', () => {
+      cy.visit('http://localhost:4200/');
+  
+      cy.get('app-products').should('exist');
+      cy.get('app-quickview').should('exist');
+      cy.get('app-basket').should('exist');
+    });
+  
+    it('should add product to the basket', () => {
+      cy.get('[id="addToBasketButton-0"]')
+        .click();
+    
+      cy.get('[id="basketTotal"]').should('not.contain', '0');
+      cy.get('[id="basketLength"]').should('not.contain', '0');
```

Comment:
Total we are not certain as price might change, but we can check how many products do we have i the basket as this quantity should be solid ( possible bug :) )